File: Searching-And-Sorting/Quick_Sort.rb
def quicksort(arr)
  return [] if arr.empty?
  pivot_element = arr.first
  left_of_pivot, right_of_pivot = arr[1..-1].partition { |num| num <= pivot_element }
  quicksort(left_of_pivot) + [pivot_element] + quicksort(right_of_pivot)
end

File: Searching-And-Sorting/quickSort.js
function quickSort(arr) {
  if (arr.length <= 1) return arr;

  let pivotElement = arr[0];
  let lessThanPivot = [];
  let greaterThanPivot = [];

  for (let i = 1; i < arr.length; i++) {
    let num = arr[i];
    if (num <= pivotElement) {
      lessThanPivot.push(num);
    }
    else {
      greaterThanPivot.push(num);
    }
  }

  return quickSort(lessThanPivot).concat([pivotElement]).concat(greaterThanPivot);
}
